backend/dal/Services/Concrete/ParcelService.cs

```diff
@@ -200,25 +201,106 @@ public Parcel Add(Parcel parcel)
         /// <returns></returns>
         public Parcel Update(Parcel parcel)
         {
-            parcel.ThrowIfNotAllowedToEdit(nameof(parcel), this.User, Permissions.PropertyEdit);
+            var existingParcel = this.Context.Parcels
+                .Include(p => p.Agency)
+                .Include(p => p.Address)
+                .Include(p => p.Evaluations)
+                .Include(p => p.Buildings).ThenInclude(b => b.Evaluations)
+                .Include(p => p.Buildings).ThenInclude(b => b.Address)
+                .SingleOrDefault(u => u.Id == parcel.Id) ?? throw new KeyNotFoundException();
 
-            var entity = this.Context.Parcels.Find(parcel.Id) ?? throw new KeyNotFoundException();
+            parcel.ThrowIfNotAllowedToEdit(nameof(parcel), this.User, "property-edit");
 
             var userAgencies = this.User.GetAgencies();
-            if (!userAgencies.Contains(entity.AgencyId)) throw new NotAuthorizedException("User may not edit parcels outside of their agency.");
+            if (!userAgencies.Contains(parcel.AgencyId)) throw new NotAuthorizedException("User may not edit parcels outside of their agency.");
 
             // Do not allow switching agencies through this method.
-            if (entity.AgencyId != parcel.AgencyId) throw new NotAuthorizedException("Parcel cannot be transferred to the specified agency.");
+            if (existingParcel.AgencyId != parcel.AgencyId) throw new NotAuthorizedException("Parcel cannot be transferred to the specified agency.");
 
             this.Context.Parcels.ThrowIfNotUnique(parcel);
 
-            this.Context.Entry(entity).CurrentValues.SetValues(parcel);
-            entity.UpdatedById = this.User.GetUserId();
-            entity.UpdatedOn = DateTime.UtcNow;
-
-            this.Context.Parcels.Update(entity);
+            //Add/Update a parcel and all child collections
```

Comment:
this entire 60ish block could be replaced by:
context.Update(rootEntity);
context.SaveChanges();
which handles "Upserting" the parent and all child entities, supporting mixing and matching adding new entities and updating existing entities. The problem is that "Upserting" does not support deletes. My original thought was that I could messily handle the deletes by loading the existing parcel and then disposing so I could use Context.Update. However, the concept of detaching entities with their children is not supported as of this time. https://github.com/dotnet/efcore/issues/18406

Only solution at this time is to not handle any kinds of delete operations in this update function.

Reply:
I think not supporting delete here makes the most sense presently.